examples/nat_topo.py

```diff
@@ -153,24 +145,47 @@ def test():
     runCmd(net, in1, "sleep 1")
     runCmd(net, in1, "examples/nat_scapy_tests.py client2", xterm=True)
     waitOutput(net, out0)
-    exitcode = int(runCmd(net, in1, "echo $?").rstrip('\n').rstrip('/r'))
-    if (exitcode != 0):
-        print "WARNING scapy test #2 failed. %d" % exitcode
+    clientExitcode = runCmd(net, in1, "echo $?").rstrip('\n\r')
+    serverExitcode = runCmd(net, out0, "echo $?").rstrip('\n\r')
+    if (clientExitcode != "0" or serverExitcode != "0"):
+        print "WARNING scapy test #2 failed. client %s, server %s" % (clientExitcode, serverExitcode)
     else:
         print "Scapy test #2 passed"
 
+    # If we couldn't show the Pax output in a separate window, show it now.
+    if not config.X_windows:
+        sendInt(net, nat0)
+        print "Show Pax output? (y/N)"
+        if (sys.stdin.read(1).upper() == "Y"):
+            waitOutput(net, nat0, verbose=True) # Print output
+        else:
+            waitOutput(net, nat0) # Don't print, just wait
+
     net.stop()
 
 # List topologies defined in this file for Mininet
 topos = { 'nat': (lambda: NatTopo())}
 
 # Helper procedures
+def setup_xterm(h, cmd, title=None):
+    if config.X_windows:
+        if title is None:
+            title = "%s> $ %s" % (h.name, cmd)
+
+        # Use a named pipe to show the output in a separate terminal, but run
+        #  the command in this shell so we have access to the exit code.
+        pipe = "/tmp/cmdpipe%d" % (random.randint(0,1000))
+        h.cmd("mkfifo %s" % pipe)
```

Comment:
Yes, in the following line there is a `rm <pipe>` after the job is done (in the xterm), but it isn't very clear. A lot of this script is pretty opaque, so I should make it clearer.